Add support for local cluster rekor server deploy

robnester-rh opened the pull request:
This adds support for deploying rekor server in the local cluster.

Michkov committed:
Create MR when new build-definitions pipelines are created

Clone infra-deployments, update the code, push and create MR.

Michkov committed:
Automatically create postgresql secret for gitops

During bootstrap generate postgresql secret to avoid post steps.

Michkov committed:
GitOps: do not duplicate files

Use manifests from the upstream project

johnmcollier committed:
Update kustomization.yaml

davidmogar committed:
Add release component

johnmcollier committed:
Merge pull request #162 from redhat-appstudio/johnmcollier-patch-3

Update HAS to f9bd765925c2cc968701f236ab13892ef17a7e58

sbose78 committed:
Support builds for private git repos (#165)

Michkov committed:
Preview mode: set docker.io globally

Avoid pull limit issue with docker.io by allow to set global image pull
secret.

Michkov committed:
Merge pull request #166 from Michkov/docker-io-workaround

Preview mode: set docker.io globally

Michkov committed:
Merge pull request #163 from davidmogar/release

Add release component

simonbaird commented:
Cool! I want to test it. I'll add a few early notes before I do that.

simonbaird commented on hack/chains/deploy_local_rekor.tpl:
```diff
@@ -0,0 +1,30 @@
+  source:
+    path: helm-charts/rekor
+    repoURL: 'https://github.com/sigstore/sigstore-helm-operator'
+    targetRevision: HEAD
```
I would have expected 'main' instead of HEAD, but I guess it would work the same, and maybe HEAD makes less assumptions, so not if there's any advantage.

simonbaird commented on hack/chains/deploy_local_rekor.sh:
```diff
@@ -0,0 +1,66 @@
+trap "rm -f $deployment_file" 0 2 3 15
+
+# sed is awesome...
+sed -e "s/\${domain}/$domain/" $input_file > $deployment_file
```
It's crude but should do the job.

I once used `envsubst` but I don't think it's particularly better.

I do wonder if argocd has a built in way to do this, but that can be a question for another day, rather than a blocker for this PR IMO.

simonbaird commented on hack/chains/deploy_local_rekor.sh:
```diff
@@ -0,0 +1,66 @@
+domain=$( kubectl get ingresses.config.openshift.io cluster -o json|jq -r ".spec.domain" )
+
+# Set the template input file
+input_file="$PWD/deploy_local_rekor.tpl"
```
The .tpl makes me think it's a standard format. Maybe a different name is better, e.g.:
* local-rekor-app-template.yml

robnester-rh replied:
`.tpl` isn't a "standard" as such, but just indicates that the file is a template of some sort. I'd shy away from using a `.yml` or `.yaml` file extension for the reason that, technically, it's not valid YAML and someone would complain.

simonbaird replied:
Ack, okay.

simonbaird commented on hack/chains/deploy_local_rekor.sh:
```diff
@@ -0,0 +1,66 @@
+  echo "See https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_login/ for details"
+  echo "Note: for CRC, use the --sso option"
+  exit 1
+fi
```
There's something similar to this in gitops-sync.sh . Maybe that is CRC specific, but perhaps we can make them more consistent with each other.

robnester-rh replied:
Thanks for the pointer, I've integrated the functionality!

simonbaird commented on hack/chains/deploy_local_rekor.sh:
```diff
@@ -0,0 +1,66 @@
+the following command:
+
+curl https://rekor-server.$domain/api/v1/log
+"""
```
In hack/bootstrap.sh there's some examples of waiting for a route. Could copy that technique if you wanna make it more user friendly.

robnester-rh replied:
awesome! I've integrated a similar function. Thanks for the suggestion.

simonbaird commented on hack/chains/deploy_local_rekor.tpl:
```diff
@@ -0,0 +1,30 @@
+  name: rekor
+spec:
+  destination:
+    name: ''
```
Why do we set name to empty string here? Maybe a comment to explain it?

simonbaird commented on hack/chains/deploy_local_rekor.sh:
```diff
@@ -0,0 +1,66 @@
+# Check the exit code from above. From what I can tell, exit code 20
+# corresponds to "you've not logged in yet". If we can figure out the
+# TODO above, then we might be able to skip this or do it better
+if [[ $(echo $?) == 20 ]]; then
```
Pretty sure this is okay, no need to echo:

    if [[ $? == 20 ]]; then

simonbaird commented on hack/chains/deploy_local_rekor.sh:
```diff
@@ -0,0 +1,66 @@
+fi
+
+# TODO: Ensure we're actually logged in before trying to execute argocd command
+argocd app create --grpc-web --upsert -f $deployment_file #2> /dev/null
```
1. I can't see it in the docs, but if argocd could take `-` to mean stdin then you could probably do a one-liner, e.g.:

    sed -e "s/\${domain}/$domain/" $input_file | argocd ... -f -


2. `#2> ...` looks wrong, is it supposed to be `2> /dev/null` or did you comment it out for testing and leave it that way.

robnester-rh replied:
Yeah, it was commented out for testing; I wasn't sure if I wanted to just trash the STDERR output or not... I'm going to drop the redirect in the next commit as the STDERR can be useful.

maysunfaisal committed:
Update HAS to include CDQ w/Alizer

Signed-off-by: Maysun J Faisal <maysunaneek@gmail.com>

johnmcollier committed:
Merge pull request #167 from maysunfaisal/CDQ-alizer-1

Update HAS to include CDQ w/Alizer

jgwest committed:
GitOps Service: Update to managed-gitop 8c72ce9bc207c389e260a2808514083accbe42cf (#168)

Michkov committed:
Merge pull request #164 from hacbs-contract/contract-devel

Tekton chains support and related scripts

Michkov committed:
Merge pull request #161 from Michkov/gitops-enhancement

Gitops enhancement

Michkov committed:
Merge pull request #150 from Michkov/build-create-mr

Create MR when new build-definitions pipelines are created

Michkov committed:
Preview: allow override HAS and SPI operator images

- Application service image can be overriden by HAS_IMAGE_REPO and HAS_IMAGE_TAG
- SPI operator can be overriden by SPI_OPERATOR_IMAGE_REPO and
  SPI_OPERATOR_IMAGE_TAG

sbose78 committed:
Enable Tekton chains' folks to view logs (#172)

* Enable Tekton chains' folks to view logs

* Update view-build-service.yaml

Michkov committed:
Merge pull request #170 from Michkov/preview-has-image

Preview: allow override HAS and SPI operator images

Michkov committed:
Build definitions update (#169)

from https://github.com/redhat-appstudio/build-definitions 08f2a927923ede328bdaa2f9a115e7fd2a8044fa

Co-authored-by: Build deployer <>

sbose78 committed:
Enable us to see what's happening in the agocd namespace (#174)

Really, looking for the Argo CD route, and the Application CRs.

zregvart commented on hack/chains/deploy_local_rekor.sh:
```diff
@@ -0,0 +1,81 @@
+fi
+
+#argocd app create --grpc-web --upsert -f $deployment_file #2> /dev/null
+echo $output | argocd app create --grpc-web --upsert -f - 
```
```suggestion
echo "$output" | argocd app create --grpc-web --upsert -f - 
```

I needed to add this on my end as `$output` contains newlines

zregvart reviewed (commented):
Other than that multi-line issue this worked on my cluster

brunoapimentel committed:
Add brunoapimentel to view HAS

Signed-off-by: Bruno Pimentel <bpimente@redhat.com>

simonbaird committed:
Contract team view permission for tekton-chains

A followup from recently merged #172.

These are the current devs in the enterprise contract team. This
change allows team members permission to inspect the tekton chains
configuration and watch the chains controller logs, which should be
useful for troubleshooting and generally looking at chains in the
stage cluster.

johnmcollier committed:
Update HAS to ebf7fed9ab85eae3ce2736f089886594d5765a39 (#176)

Pulls in:
- https://github.com/redhat-appstudio/application-service/pull/77
- https://github.com/redhat-appstudio/application-service/pull/76
- https://github.com/redhat-appstudio/application-service/pull/75

Commit:
Build definitions update

from https://github.com/redhat-appstudio/build-definitions 8201a567956ba6d2095d615ea2c0f6ab35f9ba5f

sbose78 committed:
Pull in changes associated with bundle fixes (#179)

https://github.com/redhat-appstudio/application-service/pull/79

Michkov committed:
Merge pull request #177 from simonbaird/contract-view-access

Contract team view permission for tekton-chains

Michkov committed:
Merge pull request #178 from redhat-appstudio-mr-creation/build-definitions-update

Build definitions update

Michkov committed:
Merge pull request #175 from brunoapimentel/add-view-has-user

Add brunoapimentel to view HAS

zregvart committed:
Signature verification for in-toto via verify-blob

For in-toto the data being signed is in the DSSE protocol format[1], so
when verifying using `cosign verify-blob` we need to present the data in
that format.

[1] https://github.com/secure-systems-lab/dsse/blob/master/protocol.md

Ref. https://issues.redhat.com/browse/HACBS-89

sbose78 committed:
Setup a tenant namespace for Red Hat ( HACBS ) (#185)

* Update kustomization.yaml

* Create redhat-namespace.yaml

sbose78 committed:
make sbose78 an admin in the org redhat (#186)

sbose78 committed:
allow sbose78 to create release strategies(hacbs) (#187)

* allow sbose78 to create ReleaseStrategies

* Update shbose.yaml

simonbaird commented:
Patch looks fine to me, nice work!

I didn't try it out yet, but aim to do that soon.

I suppose it will need another PR for main branch in the main repo since we retired contract-devel.

sbose78 committed:
enable management of gitopsdeployments (#189)

Michkov committed:
Merge pull request #183 from zregvart/pr/dsse-protocol-for-in-toto-format

Signature verification for in-toto via verify-blob

simonbaird committed:
Merge pull request #182 from zregvart/pr/add-service-ca

Add service CA to the chains-ca-cert ConfigMap

Michkov committed:
Build definitions update (#184)

from https://github.com/redhat-appstudio/build-definitions 93dac682ee16a08634d593b836758e1d4fa8d967

Co-authored-by: Build deployer <>

sbose78 committed:
See what's going on in Parag's namespace (#190)

robnester-rh committed:
HACBS60 - Add support for local cluster rekor

This adds support for deploying rekor server in a local cluster.

Signed-off-by: Rob Nester <rnester@redhat.com>
